### DISCUSSION TITLE
Why FastAPI-Users?

### DISCUSSION THREAD

**Discussion by seperman** [fastapi-users/fastapi-users#656] [General]
Hello,

While I was going through the FastAPI-Users docs I happened to see what already comes with [FastAPI itself](https://fastapi.tiangolo.com/tutorial/security/first-steps/) and now I'm confused what exactly FastAPI-Users provides that is not already provided by FastAPI? If that could be explained in the docs, that would be great. It makes it difficult to decide to use this library when it is not clear what we gain from it. I know you have a list of features on the docs "about" page but again, many of those are covered by FastAPI itself already in the link I posted.

Thank you!

**Comment by frankie567** [comment]
Hi @seperman,

What shows the FastAPI documentation page you link is how you can implement an authentication system yourself from scratch; having to take care about the registration route, password hashing, JWT generation and verification, etc.

What we try to propose with FastAPI Users is a plug-and-play library with all this logic ready. If you follow [one of the full example we propose](https://frankie567.github.io/fastapi-users/configuration/full-example.html#sqlalchemy), you'll get, among other things:

* Registration routes
* Authentication using authentication header or cookie
* Forgot/reset password logic
* E-mail validation logic

In a just a few lines of code and without having to worry about more low-level things.

If you prefer to have full-control of your authentication system, you can of course implement your own using the official guidelines from FastAPI documentation. FastAPI Users is just a shortcut to have a tested/production-ready system in minutes.

Cheers!
